Tasks/Kubernetes/src/clusterconnection.ts

```diff
@@ -24,8 +24,21 @@ export default class ClusterConnection {
         {
             tl.debug(tl.loc("DownloadingClient"));
             this.kubectlPath = path.join(this.userDir, "kubectl") + this.getExecutableExtention();
-            var version = await utils.getStableKubectlVersion();
-            await utils.downloadKubectl(version, this.kubectlPath);
+            let versionOrLocation = tl.getInput("versionOrLocation");
+			if(versionOrLocation === "version")
+			{
+				let versionSpec = tl.getInput("versionSpec");
+				let checkLatest: boolean = tl.getBoolInput('checkLatest', false);
+				let version  = await utils.getKubectlVersion(versionSpec, checkLatest, this.kubectlPath);
+				await utils.downloadKubectl(version, this.kubectlPath);
+			}
+			else if( versionOrLocation === "location")
+			{
+				let pathToKubectl = tl.getPathInput("specifyLocation", false, true);
```

Comment:
When location is provided just check if the file exists and change the value of this.kubectlPath to specifyLocation and do chmod. This way we don't have to copy to a location. Secondly, it has a benefit that if kubectl is corrupt user will come to know in the log and will not blame us for using any other kubectl of our choice.